### PR TITLE
Closes #2383 Upgrade tag page view.

### LINKED ISSUE #2383: Configure how taxonomy term pages should display
View: category pages
The categories for each article is selected when editing the page: https://www.screencast.com/t/Lu39IC4Az3
Screenshot: https://www.screencast.com/t/MYpq1dPYnxU
Example location: https://news.arizona.edu/arts

This open issue should provide this view: https://github.com/az-digital/az_quickstart/issues/735
Adding the needs of Comms to this ticket will be helpful.
- Cameron

Link to full News & UA@Work Content Types audit: https://docs.google.com/presentation/d/1hdlMg5gv_h09DvoCiErIOMu-mtc3M402GFAMcnGz4Ko/edit?usp=sharing

### PR DESCRIPTION
**Pull request by Krystation**
idk whats up but i cant add a comment on here without getting an awww snap page so i'll leave this here:
First commit to this, only added a new page view and made a display mode
So, I just changed these 2 files. Let me know if more needs to be changed

## Description

Created a new page view under news/taxonomy_term and a new list display mode for it as well

## Related issues
#2383

## How to test

Ensure page view is there and go to /news/some_term should only display relevant ones

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [x] Update experimental module
- **Minor release changes**
   - [x ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [x] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_category_view.yml [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+    - core.entity_view_mode.node.az_category_view
+    - field.field.node.az_news.field_az_attachments
+    - field.field.node.az_news.field_az_body
+    - field.field.node.az_news.field_az_byline
+    - field.field.node.az_news.field_az_contacts
+    - field.field.node.az_news.field_az_expiration_date
+    - field.field.node.az_news.field_az_link
+    - field.field.node.az_news.field_az_main_content
+    - field.field.node.az_news.field_az_media_image
+    - field.field.node.az_news.field_az_metatag
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_category_view.yml:14]:
```suggestion
```

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_category_view.yml [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+hidden:
+  az_news_read_more: true
+  field_az_attachments: true
+  field_az_body: true
+  field_az_byline: true
+  field_az_contacts: true
+  field_az_expiration_date: true
+  field_az_link: true
+  field_az_main_content: true
+  field_az_metatag: true
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_category_view.yml:386]:
```suggestion
```

File: modules/custom/az_news/config/install/views.view.az_news.yml [hunk 3]
```diff
@@ -26,94 +27,34 @@ base_table: node_field_data
 base_field: nid
 display:
   default:
-    display_plugin: default
     id: default
     display_title: Master
+    display_plugin: default
     position: 0
     display_options:
-      access:
```
Comment (joshuasosa) [inline modules/custom/az_news/config/install/views.view.az_news.yml:34]: Lots of changes to Master display, should probably be avoided.

**Review by joshuasosa — COMMENTED** [review]
Viewing this on the Probo build, the styling looks off:
https://524836a0-52fb-492c-ae73-f0e0bfc91074.probo.build/news/research
image

Styling should match up with how it looks on /news.

Will associated taxonomy tags be added to a display? I don't see them added to the news list nor individual news article pages, so I'm not sure how to get to a taxonomy page added from this PR except by manually typing in the address.

**Comment by trackleft** [comment]
Generally, this looks OK to me. Few suggestions.

On news.arizona.edu there is a "tags page" view that overrides /taxonomy/term/% for the entire site, and all taxonomy terms (not just news tags). We should consider introducing this similar functionality to Quickstart via the `az_core` module.

image

This is generally controlled via the path settings for specific terms.
Examples:
The arts and humanities taxonomy term, under categories has the `arts` alias

image

So if I go to `/arts` I see the view that overrides `/taxonomy/term/%`

Another example: 2020 Election  is a tag, not a category, an has a different alias pattern: `tags/2020-election`

However, the same view is being called.


Since AZ Quickstart comes with the same view as the Drupal 7 taxonomy/term/% view at this location `/admin/structure/views/view/taxonomy_term`

You could probably clone that, and change the name to create this.

File: modules/custom/az_news/config/install/views.view.az_news.yml [hunk 4]
```diff
@@ -581,31 +582,174 @@ display:
...
+          entity_type: taxonomy_term
+          entity_field: name
+          plugin_id: string
+          default_action: default
+          exception:
+            value: all
+            title_enable: false
+            title: All
+          title_enable: true
+          title: '{{ path(''<current>'')|split(''/'')|last|replace({''-'': '' ''})|capitalize }}'
```
Comment (joshuasosa) [inline modules/custom/az_news/config/install/views.view.az_news.yml:620]: This twig syntax seems to work fine but might fail if the taxonomy term itself includes a hyphen.

Consider using this module:
https://www.drupal.org/project/views_taxonomy_term_name_into_id

Possibly also including this patch for the module:
https://www.drupal.org/project/views_taxonomy_term_name_into_id/issues/3092498

Comment (Krystation) [reply modules/custom/az_news/config/install/views.view.az_news.yml:620]: Sweet, I'll redo this the way CG recommended, I'll look into this module. Thanks!

**Comment by Krystation** [comment]
@trackleft Thanks for the alternative. I'll redo that way :)

File: modules/custom/az_news/config/install/views.view.az_news.yml (hunk 4 above)
Comment (trackleft) [reply modules/custom/az_news/config/install/views.view.az_news.yml:620]: @joshuasosa nice find with that module and patch, good point about hyphenated term names. @Krystation that is some fancy twig

**Comment by ewlyman** [comment]
Question: where are we getting the categories to test this? 
I'm assuming they aren't configured on QS main branch.
Krys, did you just make up some tags to check your display?

**Comment by Krystation** [comment]
Yo! okay so i undid what i did before. And followed CGs suggestion on cloning the taxonomy term PR. Still kept the display mode though.

**Comment by trackleft** [comment]
Ok, so it looks like the only content type display mode(entity display mode) shared between all of the default content types is `az_card` so we should use that, or enable the `az_row` display mode for news so that they can all be displayed using this view.

**Comment by trackleft** [comment]
Are there any updates on this, should we reassign this, so we can have it done before friday?

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_row.yml [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    enabled: true
+    settings:
+      smart_title__link: false
+      smart_title__tag: h3
+      smart_title__classes:
+        - text-midnight
+        - my-0
+        - font-weight-bold
+_core:
+  default_config_hash: Phsrij3O1JJ05W5fTc_L_X0QMuHuDI_OK2cJN5o8IA4
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_row.yml:156]:
```suggestion
```

**Comment by trackleft** [comment]
See 
https://6ad95e9b-1fab-47a0-bcd5-743e2cfd11bf.probo.build/event-categories/athletics
https://6ad95e9b-1fab-47a0-bcd5-743e2cfd11bf.probo.build/news-tags/research
https://6ad95e9b-1fab-47a0-bcd5-743e2cfd11bf.probo.build/page-categories/no-sidebar
https://6ad95e9b-1fab-47a0-bcd5-743e2cfd11bf.probo.build/person-categories/research-associates

**Comment by trackleft** [comment]
FYI, existing markup for row on news.arizona.edu is this
```
	<a class="remove-external-link-icon list-link-block" href="https://news.arizona.edu/story/arizona-football-jerseys-find-new-life-apparel-designed-students">
		<div class="media">
			<div class="media-body">
				<div class="text-muted">
					Wednesday
				</div><strong class="sans media-heading text-size-h3">Arizona football jerseys find new life in apparel designed by students</strong>
				<p>The Arizona Replay clothing line <span><span><span>features designs inspired by University of Arizona fashion student creations that combine athletic gear with vintage clothing.&nbsp;</span></span></span></p>
			</div>
			<div class="required-fields media-right"><img alt="University of Arizona students model clothing and accesories from REPLAY Arizona, a vintage fashion line" height="100" src="https://cdn.uanews.arizona.edu/s3fs-public/styles/uanews_thumbnail/public/AMP02850-Enhanced.jpg?itok=V7_zlko8" title="University of Arizona students model clothing and accesories from REPLAY Arizona, a vintage fashion line" typeof="foaf:Image" width="100"></div>
		</div>
	</a>
```

**Comment by camikazegreen** [comment]
- All of the filtering looks like it is working correctly.
- The displays for Events, Pages and Person look great.
- The display for News looks off. I agree with @joshuasosa that it should mirror what you would see on /news, but filtered to the taxonomy term.
- On all of the displays, the RSS link looks out of place

**Review by bberndt-uaz — APPROVED** [review]

**Comment by joeparsons** [comment]
We decided in our 2023-06-16 Friday meeting to go ahead and merge this so it can be included in our Quickstart 2.7.0 pre-release and get tested during the extensive pre-release testing that is planned.  If issues are discovered during testing, we may decide to revert this change and re-open the issue to fix this in a future release.

**Review by trackleft — APPROVED** [review]

**Pull request merged**